internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec.go

```diff
@@ -260,7 +260,7 @@ func verifyOutputURL(output *loggingv1.OutputSpec, conds loggingv1.NamedConditio
 	if output.URL == "" {
 		// Some output types allow a missing URL
 		// TODO (alanconway) move output-specific valiation to the output implementation.
-		if output.Type == loggingv1.OutputTypeKafka || output.Type == loggingv1.OutputTypeCloudwatch ||
+		if output.Type == loggingv1.OutputTypeCloudwatch ||
```

Comment:
@alanconway I don't believe this is the correct fix.  The original design essentially calls for a kafka configuration to have at least one broker entry instead of a URL.  Shouldn't the fix be to verify kafka has one broker

Reply:
This is what I went by: https://github.com/openshift/cluster-logging-operator/blob/eb22dc222ff11c7f368801f78ca43176c2b65596/apis/logging/v1/output_types.go#L174
// Brokers specifies the list of brokers
// to register in addition to the main output URL

Reply:
I defer to @alanconway here to approve and release the hold

Reply:
The original design was that kafka could have a URL **or**  a broker entry, so the "kafka" part is only necessary if you need multiple target brokers. Here's the API doc, which could definitely be improved: 
```
// Brokers returns the list of broker endpoints of a kafka cluster.
// The list represents only the initial set used by the collector's kafka client for the
// first connention only. The collector's kafka client fetches constantly an updated list
// from kafka. These updates are not reconciled back to the collector configuration.
// The list of brokers are populated from the Kafka OutputSpec `Brokers` field, a list of
// valid URLs. If none provided the target URL from the OutputSpec is used as fallback.
// Finally, if neither approach works the current collector process will be terminated.
```

Reply:
To state it more clearly: the total set of URLs for kafka should include all of
- outputs.url if present
- kafka.brokers if present
Either of outputs.url or kafka.brokers can be missing or empty, but not both.

Reply:
@alanconway The design doc (could you share a link?) sounds like `kafka.brokers` takes precedence over `outputs.url`. 
 I'm assuming we need to generate TLS config if we see at least one TLS URL in the mix?

Reply:
You're right about the API doc, but I am inclined to change the API doc to what I said above. If there's a brokers entry the URL is optional but I think it will be confusing if the URL is included but ignored. +1 on TLS configuration if there are any TLS URLs.